SDK PR for Add enableCrossZoneUpgrade and prioritizeUnhealthyInstances properties to RollingUpgradePolicy for VMSS

kimberlyyipc opened the pull request:
REST API Pull Request: https://github.com/Azure/azure-rest-api-specs/pull/11184

Commit:
SDK PR for VMSS Rolling upgrade policy changes

markcowl reviewed (changes requested):
The code changes look OK, but there are some things that are missing:
- Please add in a reference to the corresponding Swagger PR for this change
- running the generation script produces an updated metadata file in the eng directory, please add this in to the PR
- When api versions change, you need to update Az.Sdk.RP.props to change the tag to match the new api-version
- You still need to re-record the tests and add in new tests for any new functionality

Also, can you coordinate with the owner of this PR: https://github.com/Azure/azure-sdk-for-net/pull/16844

Commit:
resolving comments

Commit:
updated metadatafile

kimberlyyipc commented:
> The code changes look OK, but there are some things that are missing:
> 
> * Please add in a reference to the corresponding Swagger PR for this change
> * running the generation script produces an updated metadata file in the eng directory, please add this in to the PR
> * When api versions change, you need to update Az.Sdk.RP.props to change the tag to match the new api-version
> * You still need to re-record the tests and add in new tests for any new functionality
> 
> Also, can you coordinate with the owner of this PR: #16844

kimberlyyipc commented:
@markcowl I've 
- added reference to the corresponding Swagger PR for this change 
- added the metadata file in the eng directory
- updated Az.Sdk.RP.props to reflect the new api-version

and cannot record/ work on new tests yet due to ARM manifest rollout

markcowl commented on sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperations.cs:
```diff
@@ -134,7 +138,7 @@ public async Task<AzureOperationResponse<Gallery>> UpdateWithHttpMessagesAsync(s
         /// A response object containing the response body and response headers.
         /// </return>
-        public async Task<AzureOperationResponse<Gallery>> GetWithHttpMessagesAsync(string resourceGroupName, string galleryName, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<AzureOperationResponse<Gallery>> GetWithHttpMessagesAsync(string resourceGroupName, string galleryName, string select = default(string), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
```
This is a breaking change - we should add an overload in customization that matches the old parameters

kimberlyyipc replied:
How is this a breaking change? Isn't this new added parameter optional?

kimberlyyipc replied:
Nvm comment above ^ I can see why it is a breaking change now. I've added some updates for this, please take a look.

markcowl commented on sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/GalleriesOperationsExtensions.cs:
```diff
@@ -150,12 +154,16 @@ public static Gallery Get(this IGalleriesOperations operations, string resourceG
             /// The cancellation token.
             /// </param>
-            public static async Task<Gallery> GetAsync(this IGalleriesOperations operations, string resourceGroupName, string galleryName, CancellationToken cancellationToken = default(CancellationToken))
+            public static async Task<Gallery> GetAsync(this IGalleriesOperations operations, string resourceGroupName, string galleryName, string select = default(string), CancellationToken cancellationToken = default(CancellationToken))
```
same - should add customization to protect from breaking change

kimberlyyipc replied:
wondering the same thing about this change

kimberlyyipc replied:
resolved

markcowl commented on sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/GalleryImageUpdate.cs:
```diff
@@ -48,22 +48,23 @@ public GalleryImageUpdate()
-        /// Image Definition.</param>
-        public GalleryImageUpdate(OperatingSystemTypes osType, OperatingSystemStateTypes osState, GalleryImageIdentifier identifier, string id = default(string), string name = default(string), string type = default(string), IDictionary<string, string> tags = default(IDictionary<string, string>), string description = default(string), string eula = default(string), string privacyStatementUri = defa ...
+        /// image definition.</param>
+        public GalleryImageUpdate(OperatingSystemTypes osType, OperatingSystemStateTypes osState, GalleryImageIdentifier identifier, string id = default(string), string name = default(string), string type = default(string), IDictionary<string, string> tags = default(IDictionary<string, string>), string description = default(string), string eula = default(string), string privacyStatementUri = defa ...
```
This is a breaking change, should add a customization with a constructor overload that matches the old version

kimberlyyipc replied:
added

markcowl commented:
@kimberlyyipc Nted a couple of places where there are breaking changes we should work around.  Otherwise, lgtm

Commit:
resolving breaking changes and added test

kimberlyyipc commented:
@markcowl I've added some updates and a new test, please take a look, thank you.

markcowl reviewed (changes requested):
You cannot fix breaking changes by editing the generated code.  Any changes you make by hand to generated code will disappear on the next generation.

Instead, Since these are partial classes, you can fix them by adding corresponding partial classes to the 'Customizations' folder, with the necessayr changes.

See this review:  https://github.com/Azure/azure-sdk-for-net/pull/16844

For examples of how to fix breaking changes in method parameters, for example.

Commit:
removing previous changes gor galleries api as we should not directly modify Generated files

markcowl commented:
@kimberlyyipc Can you please fix the merge conflicts?

kimberlyyipc committed:
Merge branch 'compute-2020-12' into compute-2020-12

kimberlyyipc commented:
> @kimberlyyipc Can you please fix the merge conflicts?

@markcowl I've resolved the merge conflicts

Sandido commented:
Closing for now.

markcowl commented:
@allenjzhang Assigned to you as this is one of the compute changes you are tracking

Commit:
resolve merge conflict
